### PR TITLE
Remove autotools build system

### PR DESCRIPTION
**Pull request by zanchey**
"With one build system, you always know whether your project will build. With two, you are never quite sure."

The autoconf-based Makefile has served us well, but it is time to part ways.

These commits cover the remaining use of autoconf and friends, add the missing lint targets to the CMake build, and finally drop the old build system.

Testing is appreciated.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -178,5 +178,20 @@ INCLUDE(cmake/Install.cmake)
...
 INCLUDE(cmake/MacApp.cmake)
 
+# Lint targets
+# This could be implemented as target properties, but the script has the useful feature of only
+# checking the currently-staged commands
+# The generator expressions below rebuild the command line for the fishlib targets
+# CMake does not support the "iquote" flag - https://gitlab.kitware.com/cmake/cmake/issues/15491
+SET(LINT_ARGS "-D$<JOIN:$<TARGET_PROPERTY:fishlib,COMPILE_DEFINITIONS>, -D>" "-I$<JOIN:$<TARGET_PROPERTY:fishlib,INCLUDE_DIRECTORIES>, -I>")
+ADD_CUSTOM_TARGET(lint
+    COMMAND echo build_tools/lint.fish -- ${LINT_ARGS}
```
Comment (faho) [inline CMakeLists.txt]: `echo`?

File: build_tools/lint.fish [hunk 2]
```diff
@@ -13,19 +13,7 @@ set all no
...
-if test $kernel_name = Linux
-    # This is an awful hack. However, the include-what-you-use program spews lots of errors like
-    #   /usr/include/unistd.h:226:10: fatal error: 'stddef.h' file not found
-    # if we don't explicitly tell it where to find the system headers on Linux. See
-    # http://stackoverflow.com/questions/19642590/libtooling-cant-find-stddef-h-nor-other-headers/
-    set -l sys_includes ($CXX -v -c src/builtin.cpp 2>&1 | \
-        sed -n -e '/^#include <...> search/,/^End of search list/s/^ *//p')[2..-2]
-    set -x CPLUS_INCLUDE_PATH (string join ':' $sys_includes)
-end
+argparse a/all -- $argv
```
Comment (faho) [inline build_tools/lint.fish]: So if you want a different CXX you'll have to use `env CXX=whatever lint.fish` instead? Yeah, that'll work.

File: CMakeLists.txt (hunk 1 above)
Comment (zanchey) [reply CMakeLists.txt]: Thanks - slipped through from testing.

File: build_tools/lint.fish (hunk 2 above)
Comment (zanchey) [reply build_tools/lint.fish]: The only use of `CXX` here was to get the system include path, but correctly set-up IWYU installations do not need this "awful hack". See include-what-you-use/include-what-you-use#100.

**Comment by zanchey** [comment]
I've tagged `last_autotools` at the commit before the old buildsystem is removed, and merged this branch along with a number of extra fixes.

**Pull request merged**